addon/components/loading-button.js

```diff
@@ -0,0 +1,41 @@
+import Ember from 'ember';
+
+const {
+  Component,
+  RSVP
+} = Ember;
+
+const LoadingButtonComponent = Component.extend({
+  tagName: 'button',
+
+  click(e) {
+    e.preventDefault();
+
+    // Because `.send` method for sending actions does not return anything, we
+    // pass it an `RSVP.defer` to be resolved in the remote action.
+    let defer = RSVP.defer();
+    defer.promise.then(() => {
+      this.$().text(this.get('originalContent'));
+    }, () => {
+
+    });
+
+    Ember.run.debounce(this, function() {
+      this.$().width(this.$().width());
```

Comment:
The button's width adapts to its content, which gives a weird effect the content is just the spinner. This just keeps the initial size while loading